### DISCUSSION TITLE
set/isElementOnFire and deprecate set/isPedOnFire

### DISCUSSION THREAD

**Discussion by CrosRoad95** [multitheftauto/mtasa-blue#337] [Ideas]
**Is your feature request related to a problem? Please describe.**
Make possibility to set elements on fire, vehicles, peds and maybe objects. There's function in gta that 

**Additional context**
CWorld::SetCarsOnFire(float,float,float,float,CEntity *) 0x5659F0

**Comment by qaisjp** [comment]
Sounds reasonable

**Comment by botder** [comment]
If you want to investigate how vehicles start burning, check out `void __thiscall CAutomobile::ProcessCarOnFireAndExplode(CAutomobile *, char)` at `0x6A7090`. Disabling that method prevents vehicles from starting the engine fire and it won't explode when you hit it against walls (health stays at 1.0), but it will explode when you shoot at it.

**Comment by MatinFree** [comment]
Different thoughts seem reasonable
